packages/front/src/components/recorder/WebcamRecorder.tsx

```diff
@@ -0,0 +1,163 @@
+import { RedoOutlined } from "@ant-design/icons";
+import { Alert, Button, Spin, Statistic } from "antd";
+import React, { MutableRefObject, RefObject, useRef, useState } from "react";
+import Webcam from "react-webcam";
+
+const { Countdown } = Statistic;
+
+enum RecorderState {
+  LOADING,
+  READY,
+  RECORDING,
+  DISPLAY,
+  ERROR,
+}
+
+const FIVE_MINUTE_IN_MS = 5 * 60 * 1000;
+
+const WebcamRecorder = () => {
+  const webcamRef: RefObject<Webcam | null> &
+    RefObject<HTMLVideoElement | null> = useRef(null);
+  const mediaRecorderRef: MutableRefObject<MediaRecorder | null> = useRef(null);
+
+  const [videoUrl, setVideoUrl] = useState("");
+  const [error, setError] = useState("");
+  const [state, setState] = useState(RecorderState.LOADING);
+
+  const handleStartRecording = () => {
+    setState(RecorderState.RECORDING);
+    mediaRecorderRef.current = new MediaRecorder(
+      webcamRef.current?.stream as MediaStream,
```

Comment:
Is it transpiled to an older version of JavaScript using something like Babel ? [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is supported from Chrome 80, which is too new.

Reply:
Transpiled to the es2019 standard as specified in the tsconfig.json. Create-react-app use a babel preset in top of that.